src/decode/xz.rs

```diff
@@ -293,14 +294,18 @@ where
     Ok(finished)
 }
 
-fn check_checksum<R>(input: &mut R, buf: &[u8], check_method: CheckMethod) -> error::Result<()>
+/// Verify block checksum against Block Check field.
+fn validate_block_check<R>(
```

Comment:
I share your sentiment, bug `Check` is the actual name of the field inside of the `Block` structure (see [section 3.4](https://tukaani.org/xz/xz-file-format-1.0.4.txt)).
Thus "Block Check" is the actual XZ nomenclature. Perhaps I can add a reference to that section so that it is clearer?

Reply:
Fine as "block_check" then.